common/djangoapps/student/views/management.py

```diff
@@ -196,7 +195,26 @@ def compose_activation_email(root_url, user, user_registration=None, route_enabl
     return msg
 
 
-def compose_and_send_activation_email(user, profile, user_registration=None):
+def _get_activation_confirmation_link(activation_key, redirect_url=None):
+    """
+    Helper function to build an activation confirmation URL given an activation_key.
+    The confirmation URL will include a "?next={redirect_url}" query if redirect_url
+    is not null.
+    """
+    root_url = configuration_helpers.get_value('LMS_ROOT_URL', settings.LMS_ROOT_URL)
+    confirmation_link = '{root_url}/activate/{activation_key}'.format(
+        root_url=root_url,
+        activation_key=activation_key,
+    )
+    if not redirect_url:
+        return confirmation_link
+
+    scheme, netloc, path, params, _, fragment = urllib.parse.urlparse(confirmation_link)
+    query = urllib.parse.urlencode({'next': redirect_url})
```

Comment:
If I recall, there was some issue in the relatively recent past for, I believe an integrated (LMS) enterprise customer, related to a redirect url that contained its own `?next=` query parameter. I'm drawing a blank on the full issue, but it had something to do with the fact that the `?next=` within the `redirect_url` itself getting url encoded caused issues when trying to do the 2nd `?next=` redirect from that was contained in the redirect url. Can try to dig up the ENT ticket related to this, if helpful.

Is this a use case we should be considering and/or supporting here?

Reply:
We talked offline and decided this wasn't a consideration here:
* The issue Adam was thinking of was problematic because of the encoded `tpa_hint` parameter.
* ...and we fixed that issue (by not redirecting Enterprise learners to the AuthN MFE)